examples/resourceServer/client/index.js

```diff
@@ -9,33 +9,54 @@ Template.authorize.onCreated(function() {
     oAuth2Server.subscribeTo.refreshTokens();
 });
 
+
+Session.setDefault("generatedSecret", "12345");
+
 Template.authorize.helpers({
-    urlParams: function() {
-        return getUrlParams();
-    },
-    isUrlParamsValid: function() {
-        var params = getUrlParams();
-        return !!params.client_id && !!params.redirect_uri && !!params.response_type;
-    },
-    grantResult: function() {
-        return grantResult.get();
-    },
-    tokenResult: function() {
-        return tokenResult.get();
-    },
-    getUserIdResult: function() {
-        return getUserIdResult.get();
-    },
-    authCodes: function() {
-        return oAuth2Server.collections.authCode.find({});
-    },
-    refreshTokens: function() {
-        return oAuth2Server.collections.refreshToken.find({});
-    }
+  generatedSecret: function (){
+    return Session.get('generatedSecret');
+  },
+  urlParams: function() {
+      return getUrlParams();
+  },
+  isUrlParamsValid: function() {
+      var params = getUrlParams();
+      return !!params.client_id && !!params.redirect_uri && !!params.response_type;
+  },
+  grantResult: function() {
+      return grantResult.get();
+  },
+  tokenResult: function() {
+      return tokenResult.get();
+  },
+  getUserIdResult: function() {
+      return getUserIdResult.get();
+  },
+  authCodes: function() {
+      return oAuth2Server.collections.authCode.find({});
+  },
+  refreshTokens: function() {
+      return oAuth2Server.collections.refreshToken.find({});
+  }
 });
 
 Template.authorize.events({
-    // Step 3.
+  // CONFIG FLOW - Step C1.1
+  'click button.addClient': function (){
+    var newClient = {
+      active: true,
+      clientId: $('input[name="clientId"]').val(),
+      redirectUri: $('input[name="redirectUri"]').val(),
+      clientSecret: $('input[name="clientSecret"]').val(),
+      clientName: $('input[name="clientName"]').val(),
+    }
+    console.log('newClient', newClient);
+    Meteor.call("oauth/addclient", newClient);
+  },
+  'click button.generateSecret': function (){
+    Session.set("generatedSecret", Random.secret());
```

Comment:
Let's keep the string as literals (single quoted) unless there is a need for the functionality of the double quotes.